src/query/generated/proto/prompb/types.proto

```diff
@@ -14,10 +14,14 @@ message Sample {
 message TimeSeries {
   repeated Label labels   = 1 [(gogoproto.nullable) = false];
   repeated Sample samples = 2 [(gogoproto.nullable) = false];
+  MetricType type         = 3;
+  string unit             = 4;
+  string help             = 5;
+
   // NB: These are custom fields that M3 uses. They start at 101 so that they
   // should never clash with prometheus fields.
-  Type type = 101;
-  Source source = 102;
+  M3Type m3_type = 101;
```

Comment:
Well thats just an unfortunate consequence of having to add `MetricType type` to match Prometheus. On the message name, see my comment above. And on the enum value name (`M3_COUNTER` etc.) - Protobuf does not allow enums sharing with the same named values in the same scope.
Long term, I think we should merge `M3Type` into `MetricType` and get rid of it (actually, only `M3_TIMER` would have to be added, with some high tag number to avoid future collision with something added by Prometheus). But that would require a phased migration of whole ingestion pipeline.
See my comments on this PR for more details.